library/src/main/scala/scala/scalajs/runtime/RuntimeString.scala

```diff
@@ -85,9 +85,11 @@ private[runtime] object RuntimeString {
 
   @inline
   def compareTo(thiz: String, anotherString: String): Int = {
+    val dynThiz = thiz.asInstanceOf[js.Dynamic]
+    val dynOther = anotherString.asInstanceOf[js.Dynamic]
+
     if (thiz.equals(anotherString)) 0
-    else if ((thiz.asInstanceOf[js.Dynamic] <
-        anotherString.asInstanceOf[js.Dynamic]).asInstanceOf[Boolean]) -1
+    else if ((dynThiz < dynOther).asInstanceOf[Boolean]) -1
     else 1
```

Comment:
I haven't verified. I'm quite certain it can, because `val dynThiz` and `val dynOther` are pure aliases, which will always be aliased away as `LocalDef`s.